test/lib/absinthe/type/custom_test.exs

```diff
@@ -0,0 +1,171 @@
+defmodule Absinthe.Type.CustomTest do
+  use Absinthe.Case, async: true
+
+  alias Absinthe.Type
+
+  defmodule TestSchema do
+    use Absinthe.Schema
+    import_types Type.Custom
+  end
+
+  @datetime %DateTime{
+    year: 2017, month: 1, day: 27,
+    hour: 20, minute: 31, second: 55,
+    time_zone: "Etc/UTC", zone_abbr: "UTC", utc_offset: 0, std_offset: 0,
+  }
+  @naive_datetime ~N[2017-01-27 20:31:55]
+  @date ~D[2017-01-27]
+  @time ~T[20:31:55]
+
+  defp serialize(type, value) do
+    TestSchema.__absinthe_type__(type)
+    |> Type.Scalar.serialize(value)
+  end
+
+  defp parse(type, value) do
+    TestSchema.__absinthe_type__(type)
+    |> Type.Scalar.parse(value)
+  end
+
+  describe ":datetime" do
+    it "serializes as an ISO8601 date and time string with UTC timezone marker" do
+      assert "2017-01-27T20:31:55Z" == serialize(:datetime, @datetime)
+    end
+
+    it "can be parsed from an ISO8601 date and time string including timezone" do
+      assert {:ok, @datetime} == parse(:datetime, "2017-01-27T20:31:55Z")
+      assert {:ok, @datetime} == parse(:datetime, "2017-01-27 20:31:55Z")
+    end
+
+    it "cannot be parsed without UTC timezone marker" do
```

Comment:
Please also include a test for parsing a non-0 offset (which should also be an error).